rosbag2/src/rosbag2/rosbag2_node.cpp

```diff
@@ -51,11 +56,83 @@ std::shared_ptr<GenericSubscription> Rosbag2Node::create_generic_subscription(
 
     get_node_topics_interface()->add_subscription(subscription, nullptr);
   } catch (const std::runtime_error & ex) {
-    RCUTILS_LOG_ERROR_NAMED(
-      "rosbag2", "Error subscribing to topic %s. Error: %s", topic.c_str(), ex.what());
+    ROSBAG2_LOG_ERROR_STREAM("Error subscribing to topic '" << topic << "'. Error: " << ex.what());
   }
 
   return subscription;
 }
+std::map<std::string, std::string> Rosbag2Node::get_topics_with_types(
+  const std::vector<std::string> & topic_names)
+{
+  std::vector<std::string> sanitized_topic_names;
+  for (const auto & topic_name : topic_names) {
+    sanitized_topic_names.push_back(topic_name[0] != '/' ? "/" + topic_name : topic_name);
```

Comment:
https://github.com/ros2/rcl/blob/master/rcl/include/rcl/expand_topic_name.h

This function does exactly what you're looking for :)

Reply:
Makes sense to use then. Thank you